tests/task_comments/test_dao.py

```diff
@@ -72,6 +72,40 @@ def test_dao_delete_comment(self):
         # Verify that task comment was deleted
         self.assertEqual(expected_response, actual_response)
 
+    def test_dao_report_violation_user_not_involved_in_relation(self):
+        self.create_task_comment()
+
+        self.assertEqual(
+            TaskCommentDAO.report_violation(user_id=3, relation_id=2, task_id=1, _id=1)[1], HTTPStatus.UNAUTHORIZED
+        )
+
+    def test_report_violation_comment_does_not_exist(self):
+        self.create_task_comment()
+
+        self.assertEqual(
+            TaskCommentDAO.report_violation(user_id=1, relation_id=2, task_id=1, _id=2)[1], HTTPStatus.NOT_FOUND
+        )
+
+    def test_report_comment_user_cant_report_own_comment(self):
+        self.create_task_comment()
+
+        self.assertEqual(
+            TaskCommentDAO.report_violation(user_id=1, relation_id=2, task_id=1, _id=1)[1], HTTPStatus.FORBIDDEN
+        )
+
+    def test_report_comment_comment_does_not_exist(self):
+        self.create_task_comment()
+
+        self.assertEqual(
+            TaskCommentDAO.report_violation(user_id=1, relation_id=2, task_id=2, _id=1)[1], HTTPStatus.NOT_FOUND
+        )
+
+    def test_report_comment_successful(self):
+        self.create_task_comment()
+
+        self.assertEqual(
+            TaskCommentDAO.report_violation(user_id=2, relation_id=2, task_id=1, _id=1)[1], HTTPStatus.OK
```

Comment:
you could also check here the message, so I suggest doing something like:
```
message, status_code = TaskCommentDAO.report_violation(user_id=2, relation_id=2, task_id=1, _id=1)
self.assertEqual(HTTPStatus.OK, status_code)
self.assertEqual(messages.APPROPRIATE_MESSAGE, message)
```

I think this works and makes this more verbose. I know most of the code uses index 🙈 but I recently learned about this, and seems to be much more clear.

however, we can leave this for other issues, perhaps newcomer ones. I know you are busy with the gsoc work.